doc/demo/sparse_train/sparse_train.md

```diff
@@ -0,0 +1,291 @@
+
+### Prepare DataProvider
+Use following dataprovider to feed DataProvider:
+```python
+# Define a py data provider
+@provider(input_types=[
+    sparse_binary_vector(18182296),
+    integer_value(2)
+])
+def process(settings, filename):
+    f = open(filename, 'r')
+    for line in f:  # read each line
+        splits = line.split(',')
+        label = int(splits[1])
+        splits.pop(0)
+        splits.pop(0)
+        sparse_non_values = []
+        for value in splits:
+            v = value.split(" ")
+            sparse_non_values.append(long(v[0]))
+        # give data to paddle.
+        yield sparse_non_values, label
+    f.close()  # close file
+```
+Please consult related DOC for ```sparse_binary_vector``` and ```integer_value```.
+
+### Prepare Model Configuration
+Here providing one simple fc_layers network, which is enough for explaining HOW to train DNN with sparse training.
+
+We use ```ParameterAttribute(sparse_update=True)``` to enable sparse training for local job as well as cluster job. The ```trainer_config_helpers``` model will set internal FLAGs automatically. At last, these layers not set with sparse FLAG do dense training while sparse layers automatically use sparse training. Generally RECOMMEND you set these layers with high sparsity input, such as first hidden layers with sparse input.
+
+The full configuration as follows:
+```python
+rom paddle.trainer_config_helpers import *
+
+label_size = 2
+data_size = 18182296
+
+""" Algorithm Configuration """
+settings(learning_rate=1e-3,
+         learning_method=MomentumOptimizer(momentum=0.9),
+         batch_size=200)
+
+""" Data Configuration """
+define_py_data_sources2(train_list='train.list',
+                        test_list=None,
+                        module='sparse_data_provider',
+                        obj='process')
+
+""" Model Configuration """
+non_value = data_layer(name='data',
+                       size=data_size)
+label = data_layer(name='label',
+                   size=label_size)
+
+hidden1 = fc_layer(input=non_value,
+                   size=128,
+                   param_attr=ParameterAttribute(sparse_update=True))
+hidden2 = fc_layer(input=hidden1,
+                   size=32)
+
+prediction = fc_layer(input=hidden2, size=label_size, act=SoftmaxActivation())
+
+outputs(classification_cost(input=prediction, label=label))
+```
+
+### Start Training 
+For local train, you do not need to do anything to start sparse training. 
+```bash
+paddle train \
+    --use_gpu=0 \
+    --config=./sparse_trainer_config.py \
+    --saving_period=1 \
+    --test_period=0 \
+    --num_passes=4 \
+    --dot_period=2 \
+    --log_period=20 \
+    --trainer_count=10 \
+    --saving_period_by_batches=5000 \
+    --local=1
+```
+For cluster train, you maybe just care about ```--ports_num_for_sparse=4``` in command line and conf.py files.  You need to understand simple cluster helper scripts in ```paddle/cluster_train/paddle.py```. Check DOC about cluster training.
```

Comment:
4 is necessary for current experiments. If it's is too little, setParameters API could generate BIG protobuf request body that is not recommended. In addition, BIG size can benefit communication performance for little size cluster JOB.